sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -611,14 +609,15 @@ class Analyzer(override val catalogManager: CatalogManager)
       val aggForResolving = h.child match {
         // For CUBE/ROLLUP expressions, to avoid resolving repeatedly, here we delete them from
         // groupingExpressions for condition resolving.
-        case a @ Aggregate(Seq(c @ Cube(groupByExprs)), _, _) =>
-          a.copy(groupingExpressions = groupByExprs)
-        case a @ Aggregate(Seq(r @ Rollup(groupByExprs)), _, _) =>
-          a.copy(groupingExpressions = groupByExprs)
-        case g: GroupingSets =>
-          Aggregate(
-            getFinalGroupByExpressions(g.selectedGroupByExprs, g.groupByExprs),
-            g.aggregations, g.child)
+        case a @ Aggregate(Seq(c @ Cube(_)), _, _) =>
```

Comment:
`c @ Cube(_)` -> `c: Cube`

Reply:
Could you add `groupingSets` in `GroupingSet` and then merge the three entries into one like this?
```
        case a @ Aggregate(Seq(c: GroupingSet), _, _) =>
          a.copy(groupingExpressions =
            getFinalGroupByExpressions(c.groupingSets, c.groupByExprs))
```